tornado/test/tcpclient_test.py

```diff
@@ -158,6 +159,17 @@ def test_source_port_fail(self):
                           '127.0.0.1',
                           source_port=1)
 
+    @gen_test
+    def test_connect_timeout(self):
+        timeout = 0.1
+
+        class TimeoutResolver(Resolver):
+            def resolve(self, *args, **kwargs):
+                return Future()  # never completes
+        with self.assertRaises(TimeoutError):
```

Comment:
There should also be a test that when a high enough timeout is used, the connection succeeds.